server/dao/products.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+const { ObjectID } = require("bson");
+const { json } = require("express");
+
+const productsCollectionName = "products";
+
+// ---------------
+// GetProducts
+// ---------------
+
+exports.getProducts = (db, category, searchString, IDs) => {
```

Comment:
It's better to have "atomic" methods in the database interface. I would split this in:
- GetProductsByIDs(ids) -> fetches all the products with the ID in the provided list. (this might be out of scope for your task, do the client already need to fetch all the products by ids?)
- FindProducts(searchString,category) -> fetches all the product which are compatible with the passed searchString and category. As you did, searchString and category can be omitted(in this case limit the maximum amount of returned products to a reasonable number). 

The handler is responsible to call the right database access method. (All the business logic is in the handlers, dais are stupid)

Reply:
I'm pretty sure you only need to implement the FindProducts dao method